### DISCUSSION TITLE
Can't sync with lan server after rejoining network

### DISCUSSION THREAD

**Discussion by Gymcap** [bitfireAT/icsx5#297] [Q&A]
I'm running an http server locally at home which exposes my generated ics file to lan. I can link to it in ICSx5 easily, but once leaving the local network (leaving my house) the sync (obviously) fails to resolve.

After coming back home and reconnecting to the network however, the sync continues to fail to resolve until I modify the entry.

I believe the app sees my link as unreachable, and sets some sort of flag to not check that link again, and It would be very useful in my usecase to be able to turn this off for this entry. If it fails to connect, id like it to try again on its next interval.

**Comment by ArnyminerZ** [comment]
This should not be happening, sync should always run, and the error should always be displayed, I believe. I will take a look and communicate what I find

**Comment by ArnyminerZ** [comment]
It should be working. Does syncing manually also fail? I mean, pulling down the screen forces the synchronization to run. Is it also failing?

**Comment by Gymcap** [comment]
Yes, it fails to reconnect until I modify the entry, like removing a character from the link, saving, and putting it back before saving again.

I'd like to reiterate that the link I'm using is only accessible when I am on my home network, and will (rightfully) fail to resolve if I'm not connected to it at sync time. But no, neither pulling to refresh nor the "synchronize now" fixes it. It seems once it fails to resolve once, it refuses to reconnect again until the entry is modified as described above.

**Comment by sunkup** [comment]
@Gymcap I this still an issue for you? I have tested it and it works just fine for me. 

> I believe the app sees my link as unreachable, and sets some sort of flag to not check that link again, and It would be very useful in my usecase to be able to turn this off for this entry. If it fails to connect, id like it to try again on its next interval.

I am not aware of ICSx5 setting any such flags. Maybe it's not ICSx5, but your android which remembers this link as unreachable? Though I have not heard of something like that before.

Would you be up for sharing your logs with us? It's not exactly easy, but we might be able to help you then.


**Here is how to get the logs:**
With [adb](https://developer.android.com/studio/command-line/adb.html), you can do powerful things with your Android device (you have to turn on USB debugging; no root required). It comes with the Android SDK, but also as a [standalone package for most operating systems](https://developer.android.com/studio/releases/platform-tools.html).

To view system logs, use [adb logcat](https://developer.android.com/studio/command-line/logcat.html). To get full ICSx⁵ logs in logcat:
   - force-quit ICSx⁵
   - do `adb shell setprop log.tag.davdroid VERBOSE` to [set the appropriate log levels](https://developer.android.com/reference/android/util/Log#isLoggable(java.lang.String,%20int))
   - do `adb logcat` again
   - start ICSx⁵ again, force-sync
